codegen/core/src/main/java/software/amazon/smithy/python/codegen/ClientGenerator.java

```diff
@@ -488,37 +490,44 @@ await sleep(retry_token.retry_delay)
         writer.popState();
 
         writer.pushState(new ResolveEndpointSection());
+        writer.addDependency(SmithyPythonDependency.SMITHY_CORE);
+        writer.addDependency(SmithyPythonDependency.SMITHY_HTTP);
+        writer.addImport("smithy_core", "URI");
+        writer.addImport("smithy_core.endpoints", "EndpointResolverParams");
+        writer.write("""
+                        # Step 7f: Invoke endpoint_resolver.resolve_endpoint
+                        context.properties["endpoint_uri"] = config.endpoint_uri
```

Comment:
How are you binding region to the context properties?  I think this won't work without that.  We could of course add all of the config properties at this point, but ideally we should limit endpoint parameters to whats required.

Reply:
> How are you binding region to the context properties?

It'll need to go into an interceptor. Of course, I'm also just dropping the config in there at the start so maybe it just makes sense to leave it there. Makes typing a bit worse though.

> ideally we should limit endpoint parameters to whats required

This is an ideal not worth pursuing. See my other comment.

Reply:
I think this is currently broken - it doesn't look like region/arbitrary config is being added to the context properties.

I don't disagree with the other comment - and I'm fine with saying everything needs to be here, but then we do need to provide the entire config through context.

Reply:
Updated to pull from config